variables.tf

```diff
@@ -46,6 +46,26 @@ variable "burstable_mode" {
   default     = null
 }
 
+variable "market_type" {
```

Comment:
Take a look at how the auto scaling group implements both of these inputs 

https://github.com/cloudposse/terraform-aws-ec2-autoscale-group?tab=readme-ov-file#input_instance_market_options

https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/blob/906f7975d4a5eaae9b40b268c988e3aa43e18e3e/variables.tf#L100-L115

Reply:
looks way more clean. 
will adjust the inputs, thanks!

Reply:
updated https://github.com/cloudposse/terraform-aws-ec2-instance/pull/202/commits/c18468768c919ad5a9b75338b3e9cd0e7b7dadeb